### PR TITLE
add optional global http session

### PR DESCRIPTION
**Pull request by theomonnom**
also fix leaked http session on plugins

### CODE REVIEW

File: livekit-agents/livekit/agents/utils/http_session.py
Comment (davidzhao) [inline livekit-agents/livekit/agents/utils/http_session.py:1]: nice!

File: livekit-agents/livekit/agents/worker.py [hunk 1]
```diff
@@ -109,30 +109,40 @@ def __init__(
...
+        self._session: aiohttp.ClientSession | None = None
+        self._api: api.LiveKitAPI | None = None
 
     async def run(self):
+        if not self._closed:
+            raise Exception("worker is already running")
+
         logger.info("starting worker", extra={"version": __version__})
+
+        # this LiveKit API object is only really useful in non-third-party workers
```
Comment (davidzhao) [inline livekit-agents/livekit/agents/worker.py]: nit: what does non third party workers mean? standard?

**Review by davidzhao — APPROVED** [review]
lgtm!

File: livekit-agents/livekit/agents/worker.py (hunk 1 above)
Comment (theomonnom) [reply livekit-agents/livekit/agents/worker.py]: yes standard

Comment (theomonnom) [reply livekit-agents/livekit/agents/worker.py]: will rename

**Pull request merged**